### ISSUE TITLE
ZIA DLP Engine Predefined Engine not found

### ISSUE THREAD

**Issue by tiwari3010** [zscaler/terraform-provider-zia#216]
Not sure if this is bug, I am opening this as a question:

Affected Resource(s)
trying to create DLP policy using predefined engine (HIPAA or GBLA). 

Use below code for getting the ID from name:

Terraform provider version:

```
terraform -version
Terraform v1.3.6
on darwin_amd64
+ provider registry.terraform.io/zscaler/zia v2.4.6
```

```
data "zia_dlp_engines" "predefinedEngineName"{
  name  = "GBLA"
}
```
Got error

```
module.zia_dlp_policies.data.zia_dlp_dictionaries.dlp_dictionaries[0]: Read complete after 3s [name=GET_SSN_KEYWORD]
╷
│ Error: no dlp engine found with name: GBLA
│ 
│   with module.zia_dlp_policies.data.zia_dlp_engines.predefinedEngineName,
│   on ../modules/data.tf line 20, in data "zia_dlp_engines" "predefinedEngineName":
│   20: data "zia_dlp_engines" "predefinedEngineName"{
│ 
╵
```
Expected Behavior
Terraform should be able to resolve predefined engine to ID to be used in policy.

Actual Behavior
Terraform is able to add custom created DLP Engines but not predefined engines.

When using the Actual ID, terraform is able to Add the engine to the policy.

**Comment by willguibr** [comment]
Hi @tiwari3010 This functionally (Search By name) is not currently supported by the provider due to API limitations. We are exploring other options to allow for search methods; however, at the moment you can search based on ID.
The recommended way currently is to retrieve the IDs via Postman and use those attributes in your HCL in the following form:

```hcl
data "zia_dlp_engines" "id" {
  id = 62
}

output "zia_dlp_engines" {
  value = data.zia_dlp_engines.id
}
```

**Comment by willguibr** [comment]
Hi @tiwari3010 We've deployed a fix for the dlp predefined engine search and it will be available in the next minor release v2.5.1 in upcoming weeks.
As for API roadmap, I am not aware of it, so I'd recommend reaching out to your TAM or Accounting team so they can check with Product management and engineering in this matter. Once the methods are available, the implementation at the provider can be completed without problems.

Regards
